src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -653,29 +653,23 @@ class UnregisteredClassesDuplicationChecker : StackObj {
   }
 };
 
-class ExcludeDumpTimeSharedClasses : StackObj {
-public:
-  bool do_entry(InstanceKlass* k, DumpTimeClassInfo& info) {
-    SystemDictionaryShared::check_for_exclusion(k, &info);
-    return true; // keep on iterating
-  }
-};
-
 void SystemDictionaryShared::check_excluded_classes() {
   assert(no_class_loading_should_happen(), "sanity");
   assert_lock_strong(DumpTimeTable_lock);
 
   if (DynamicDumpSharedSpaces) {
     // Do this first -- if a base class is excluded due to duplication,
-    // all of its subclasses will also be excluded by ExcludeDumpTimeSharedClasses
+    // all of its subclasses will also be excluded.
     ResourceMark rm;
     UnregisteredClassesDuplicationChecker dup_checker;
     _dumptime_table->iterate(&dup_checker);
```

Comment:
Which iterate function does this call?

Reply:
I changed it to `_dumptime_table->iterate_all_live_classes()`